src/main/java/com/connexta/store/service/impl/StoreServiceImpl.java

```diff
@@ -102,4 +109,28 @@ public void addIrm(
             .build(datasetId));
     log.info("Successfully indexed datasetId={}", datasetId);
   }
+
+  /* TODO: Implement the logic call promote/quarantine. Doesn't have to be a lambda. Could be a class the implements Runnable or Callable. */
+  private void poll(URI uri) {
+
+    Future<StatusResponse> pollFuture = statusService.poll(uri);
+
+    // Run on another thread so processing is not blocked
+    executorService.submit(
```

Comment:
❗️I don't think this should be asynchronous. If submitting the poll fails, we should fail the entire request. I also don't think promote and quarantine should be handled here. Sending the promote/quarantine should happen in the AttemptResult. I think this method can be as simple as. 
```java
private void poll(@NotNull final URI uri) {
  statusService.poll(uri);
}
```

This also simplifies the StatusService#poll method because it shouldn't return anything
```java
public void poll(@NotNull final URI uri) {
```

Reply:
I think this is okay to be asynchronous. From replications perspective, they don't care if polling submission fails or not, they are not going to be able to fix that error as a client. They care only that we have successfully staged the files requested to be processed. If a user is able to fix it, then they will have to view the failed logs, modify the data and then replication will attempt to stage the data again.

Reply:
It cannot be asynchronous because if we fail to kick off polling, all information will be completely lost because we already returned a success for the ingest request. Additionally, it doesn't make sense to pull the code to call the promote and quarantine endpoints asynchronously from the Attempt because we will need to send the delete request to transform. Let's talk in person to make sure we're on the same page about this.